bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java

```diff
@@ -133,6 +134,13 @@ public final class GC extends Resource {
  */
 public GC(Drawable drawable) {
 	this(drawable, SWT.NONE);
+	if(drawable instanceof Widget) {
+		data.deviceZoom = ((Widget) drawable).zoom;
+		if (drawable instanceof Control)
+			data.nativeDeviceZoom = ((Control) drawable).getShell().nativeZoom;
```

Comment:
can there be an issue, when we don't set nativeDeviceZoom with Widget and Image? We should be sure about that